app/src/main/java/app/musikus/ui/activesession/ActiveSessionViewModel.kt

```diff
@@ -276,30 +279,38 @@ class ActiveSessionViewModel @Inject constructor(
     )
 
 
+    /** ############  The final public UI state ############ */
+
     val uiState = combine(
         libraryCardUiState,
-        sessionServiceState,
+        sessionState,
         addLibraryItemDialogUiState,
-        totalDurationRoundedDownToNextSecond,
-        dialogUiState
-    ) { libraryCardUiState, sessionState, addItemDialogUiState, totalDuration, dialogUiState ->
+        dialogUiState,
+    ) { libraryCardUiState, sessionState, addItemDialogUiState, dialogUiState ->
         ActiveSessionUiState(
             cardUiStates = listOf(
                 libraryCardUiState,
                 _recorderUiState,
                 _metronomeUiState
             ),
-            totalSessionDuration = totalDuration,
-            totalBreakDuration = sessionState.pauseDuration,
-            isPaused = sessionState.isPaused,
+            totalSessionDuration = sessionState?.sessionDuration ?: 0.seconds,
+            ongoingPauseDuration = sessionState?.ongoingPauseDuration ?: 0.seconds,
+            isPaused = sessionState?.isPaused ?: false,
             addItemDialogUiState = addItemDialogUiState,
-            sections = sessionState.sections.reversed().map { section ->
-                ActiveSessionSectionListItemUiState(
-                    id = section.id,    // also used as id
+            sections = sessionState?.sections?.reversed()?.map { section ->
+                    ActiveSessionSectionListItemUiState(
+                    id = section.id,
                     libraryItem = section.libraryItem,
-                    duration = section.duration ?: sessionState.currentSectionDuration
+                    duration = section.duration
+                )
+            } ?: emptyList(),
+            runningSection = if(sessionState != null) {
```

Comment:
if you use `sessionState?.let` you can save yourself the `else null` in the end.